src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -138,7 +142,8 @@ private static void verifyLabel(Object label) {
         Class<?> labelClass = label.getClass();
         if (labelClass != Class.class &&
             labelClass != String.class &&
-            labelClass != Integer.class) {
+            labelClass != Integer.class &&
+            labelClass != EnumDesc.class) {
             throw new IllegalArgumentException("label with illegal type found: " + label.getClass());
         }
     }
```

Comment:
side are there any cases for which the `startIndex` argument passed to method `doTypeSwitch` is not `0`?